blogsy/app/controllers/user_controller.rb

```diff
@@ -18,4 +18,17 @@ def create
     end
   end
 
+  # should return a JSON representation
+  # of the posts written by a user with
+  # an username of params[:id]
+  def showposts
+    user = User.find(params[:id]).posts
+    binding.pry
+    if user
+      render json: user
+    else
+      render json: user.errors.full_messages
```

Comment:
We're not saving a user here, so there likely won't be any errors (or at least, not this type)